src/main.rs

```diff
@@ -82,85 +121,34 @@ fn main() {
   }
 }
 
-fn build_sql_opts() -> Opts {
-  let sql_vals: SqlConfig = get_config().mysql;
-  let mut builder = OptsBuilder::new();
-  builder
-    .ip_or_hostname(Some(sql_vals.endpoint))
-    .tcp_port(sql_vals.port)
-    .user(Some(sql_vals.username))
-    .pass(Some(sql_vals.password))
-    .db_name(Some(sql_vals.database));
-  builder.into()
-}
+fn add_accounts(discordid: u64, mc_user: &MinecraftUser) -> QueryResult<usize> {
+  use self::schema::minecrafters;
 
-fn add_accounts(discord_id: u64, mc_user: &MinecraftUser) -> u16 {
-  let pool: mysql::Pool = mysql::Pool::new(build_sql_opts()).unwrap();
-  // Prepare the SQL statement
-  let mut stmt: mysql::Stmt = pool
-    .prepare(
-      r"
-        INSERT INTO minecrafters
-          (discord_id, minecraft_uuid, minecraft_name)
-        VALUES
-          (:discord_id, :minecraft_uuid, :minecraft_name)
-      ",
-    )
-    .unwrap();
-  // Execute the statement with vals
-  let ret = stmt.execute(params! {
-    "discord_id" => &discord_id,
-    "minecraft_uuid" => &mc_user.id,
-    "minecraft_name" => &mc_user.name
-  });
-
-  // This code is a nightmare, undocumented as well
-  match ret {
-    Ok(_val) => 0,
-    Err(MySqlError(e)) => {
-      if e.message.contains("Duplicate entry") {
-        return e.code + 1;
-      }
-      e.code
-    }
-    Err(e) => {
-      println!("SQL FAILURE: {}", e);
-      1
-    }
-  }
-}
+  let connection = POOL.get().unwrap();
```

Comment:
Connection error is not handled here. Will thread panic without notifying the requesting user.